error: incompatible_types (See https://fb.me/eqwalizer_errors#incompatible_types)
   ┌─ check/src/funs2.erl:32:20
   │
32 │     fun(_, Acc) -> [Acc] end,
   │                    ^^^^^
   │                    │
   │                    [Acc].
Expression has type:   [[[[]]]]
Context expected type: [[[]]]
   │                    

Because in the expression's type:
  [
    [
      [
        Here the type is:     []
        Context expects type: none()
      ]
    ]
  ]

------------------------------ Detailed message ------------------------------

  [[[[]]]] is not compatible with [[[]]]
  because
  [[[]]] is not compatible with [[]]
  because
  [[]] is not compatible with []

error: incompatible_types (See https://fb.me/eqwalizer_errors#incompatible_types)
   ┌─ check/src/funs2.erl:54:5
   │
54 │     X.
   │     ^ X.
Expression has type:   [dynamic()]
Context expected type: number()

error: clause_not_covered (See https://fb.me/eqwalizer_errors#clause_not_covered)
   ┌─ check/src/funs2.erl:95:1
   │
95 │ generic({}, {}, {}) -> nok.
   │ ^^^^^^^^^^^^^^^^^^^^^^^^^^ Clause is not covered by spec

error: incompatible_types (See https://fb.me/eqwalizer_errors#incompatible_types)
    ┌─ check/src/funs2.erl:101:9
    │
101 │         fun erlang:atom_to_list/1,
    │         ^^^^^^^^^^^^^^^^^^^^^^^^^
    │         │
    │         erlang:atom_to_list/1.
Expression has type:   fun((atom()) -> string())
Context expected type: fun((atom()) -> atom())
    │         

Because in the expression's type:
  fun((atom()) ->
    Here the type is:     string()
    Context expects type: atom()
  )

------------------------------ Detailed message ------------------------------

  fun((atom()) -> string()) is not compatible with fun((atom()) -> atom())
  because
  string() is not compatible with atom()

error: incompatible_types (See https://fb.me/eqwalizer_errors#incompatible_types)
    ┌─ check/src/funs2.erl:152:27
    │
152 │             (X, X) -> X + A
    │                           ^ A.
Expression has type:   atom()
Context expected type: number()

5 ERRORS
